deploy/cdk_exec_policy/cdkExecPolicy.yaml

```diff
@@ -0,0 +1,280 @@
+AWSTemplateFormatVersion: 2010-09-09
+Description: Custom least privilege IAM policy for linking environments to dataall
+Parameters:
+  AwsAccountId:
+    Description: AWS AccountId of the account that we wish to link.
+    Type: String
+  PolicyName:
+    Description: IAM policy name (The same name must be used during CDK bootstrapping. Default is DataAllCustomCDKPolicy.)
+    Type: String
+    Default: 'DataAllCustomCDKPolicy'
+  EnvironmentResourcePrefix:
+    Description: The resource prefix value of the dataall environment. It MUST match the resource prefix that we use when we create the environment.
+    Type: String
+Resources:
+  CDKCustomExecutionPolicy0:
+    Type: 'AWS::IAM::ManagedPolicy'
+    Properties:
+      ManagedPolicyName: !Ref PolicyName
+      PolicyDocument:
+        Version: 2012-10-17
+        Statement:
+          - Sid: KMS
+            Action:
+              - 'kms:CreateKey'
+            Effect: Allow
+            Resource: '*'
+          - Sid: ec2
+            Action:
+              - 'ec2:DescribeSecurityGroups'
+              - 'ec2:CreateSecurityGroup'
+              - 'ec2:createTags'
+              - 'ec2:DeleteSecurityGroup'
+              - 'ec2:RevokeSecurityGroupEgress'
+              - 'ec2:AuthorizeSecurityGroupIngress'
+              - 'ec2:AuthorizeSecurityGroupEgress'
```

Comment:
When testing creating a new Environment with no default VPC and ML Studio Enabled, required some additional permissions for stack creation:
```
                "ec2:RevokeSecurityGroupIngress",
                "ec2:AttachInternetGateway",
                "ec2:DetachInternetGateway",
                "ec2:DescribeInternetGateways",
                "ec2:DescribeRouteTables",
                "ec2:AssociateRouteTable",
                "ec2:DisassociateRouteTable",
                "ec2:CreateRoute",
                "ec2:DeleteRoute"
```


NOTE: Adding the above additional permissions may cause a character max length exceeded in the policy. I will suggest 1 area where I think we can consolidate permissions in the YAML File 

In the long-term I imagine we will have to split this out into multiple policies and pass a list of execution policies to the CDK bootstrap command as we refine this permission set